### PR TITLE
Add eslint tsdoc with fixes - w 7198851

### PR DESCRIPTION
**Pull request by bredmond-sf**
As a Commerce Cloud app developer I want to have detailed method information in my TSDOC for the generated methods

* Docstrings should have a description, parameters (including type where applicable, even when “self-explanatory”), and return
* TSDOC should take description from RAML file wherever applicable.
* SDK specific parameters can be hard coded.
* TSDOC should output associated method
* Utilize eslint TSDOC plugin

Before:
Screen Shot 2020-05-14 at 10 47 37 AM

After (including typescript sig for reference):
Screen Shot 2020-05-13 at 12 49 15 AM
Screen Shot 2020-05-13 at 12 49 30 AM

### CODE REVIEW

File: packages/generator/test/template-helpers-format-for-tsdoc.test.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+"use strict";
+
+import { expect } from "chai";
+
+import { formatForTsDoc } from "../src/templateHelpers";
+
+describe("Test formatForTsDoc template help function", () => {
```
Comment (wjhsf) [inline packages/generator/test/template-helpers-format-for-tsdoc.test.ts:13]: The first half of these tests are `.to.eq()` and the second half are `.to.be.eq()` :|

Comment (bredmond-sf) [reply packages/generator/test/template-helpers-format-for-tsdoc.test.ts:13]: good catch

File: packages/generator/src/renderer.ts [hunk 2]
```diff
@@ -166,9 +182,9 @@ function createHelpers(config: any): string {
 
 /**
  * Render the API Clients markdown file using the Handlebars template
- * @param {Map<string, WebApiBaseUnit[]>} apiFamilyMap
- * @param {Object.<string, RestApi[]} apiFamilyConfig
- * @returns {string} The rendered template
+ * @param apiFamilyMap -
+ * @param apiFamilyConfig -
```
Comment (gkhunger-sf) [inline packages/generator/src/renderer.ts:186]: Missing description

Comment (bredmond-sf) [reply packages/generator/src/renderer.ts:186]: Yea, no one responded to me when I asked about this one in our chat :) I'm happy to add one, I just don't know what it is.

Comment (monkey-jeff) [reply packages/generator/src/renderer.ts:186]: What do you think the description should be?

Comment (wjhsf) [reply packages/generator/src/renderer.ts:186]: That's why you git blame and @ping. 😛 I was going to reply, but forgot!

```
* @param apiFamilyMap - Collection of API names and the AMF models associated with each API
* @param apiFamilyConfig - The API family config
```

Comment (bredmond-sf) [reply packages/generator/src/renderer.ts]: done

**Pull request merged**